Modix.Data/Repositories/InfractionRepository.cs

```diff
@@ -267,6 +277,33 @@ public async Task<bool> TryRescindAsync(long infractionId, ulong rescindedById)
             return true;
         }
 
+        /// <inheritdoc />
+        public async Task<bool> TryRestoreAsync(long infractionId, ulong restoredById)
+        {
+            var entity = await ModixContext.Infractions
+                .Where(x => x.Id == infractionId)
+                .FirstOrDefaultAsync();
+
+            if ((entity == null) || (entity.RescindActionId != null))
+            {
+                return false;
+            }
+
+            entity.RestoreAction = new ModerationActionEntity
+            {
+                GuildId = entity.GuildId,
+                Type = ModerationActionType.InfractionRestored,
+                Created = DateTimeOffset.Now,
+                CreatedById = restoredById,
+                InfractionId = entity.Id
+            };
+
+            await ModixContext.SaveChangesAsync();
+            await RaiseModerationActionCreatedAsync(entity.RescindAction);
```

Comment:
`await RaiseModerationActionCreatedAsync(entity.RescindAction);` appears incorrect. consider replacing it with `await RaiseModerationActionCreatedAsync(entity.RestoreAction);`

Reply:
Fixed